lib/ukdebug/gdbstub.c

```diff
@@ -71,6 +72,33 @@ static __u16 gdb_backing_console_id = GDB_BACKING_CONSOLE_NONE;
 UK_LIBPARAM_PARAM_ALIAS(gdbcon, &gdb_backing_console_id, __u16,
 			"gdb backing console ID");
 
+/* This is a tristate value controlling the behavior of the gdb stub's virtual
+ * console. The virtual console implements a console device for `ukconsole`.
+ * This console device is called "virtual" because it's not a driver for a
+ * real console device. The virtual console sends all console output to the
+ * gdb host using the 'O' packet. The backing console is used to transmit this
+ * packet like any other. With the virtual console, the backing console can be
+ * used for kernel console output and for gdb communication at the same time.
+ *
+ * The value of this tristate is interpreted as follows:
+ * - (> 0): always register the virtual console
+ * - (0): never register the virtual console
+ * - (< 0): register the virtual console if the backing console has flags set
+ */
+static int gdb_virt_console_mode = -1;
+UK_LIBPARAM_PARAM_ALIAS(gdbmode, &gdb_virt_console_mode, int,
+			"-1=auto, 0=not shared 1=shared");
+
+static inline int gdb_virt_console_on(void)
```

Comment:
Somehow, I find the naming of these configs checks not ideal when reading the function that uses those later. What would you think of:
`gdb_vcon_want_shared()` and `gdb_vcon_want_auto()`?

Reply:
How about `gdb_mode_is_shared`/`gdb_mode_is_auto`?

Reply:
Good 👍🏻